File: String Manipulation/Character Removal/Python/character_removal.py
"""
    Remove all occurences of char to_remove from string s
"""
def remove_character(s, to_remove):    
    result = ""
    for c in s:
        if c != to_remove:
            result += c
    return result

if __name__ == "__main__":
    assert(remove_character("Hello World", 'o') == "Hell Wrld")
    assert(remove_character("Hello World", 'l') == "Heo Word")
    assert(remove_character("Hello World", 'a') == "Hello World")
    assert(remove_character("", 'a') == "")
    assert(remove_character("Hello World", '') == "Hello World")

File: String Manipulation/Character Removal/C++/CharacterRemoval.cpp
// CharacterRemoval.cpp
// Implementation of function used to remove all occurences of a character from a string in C++
// Created by Efren Aguilar for the codezilla project

//#include "stdafx.h"

void removeCharsFromStr(char* str, char toRemove, int strLen)
{
	int numRemoved = 0;
	for (int i = 0; i < strLen; i++)
	{
		if (str[i] == toRemove)
		{
			numRemoved++;
			//Shift string down by one character
			for (int start = i; start < strLen - numRemoved; start++)
				str[start] = str[start+1];
			str[strLen - numRemoved] = '\0';

			//String shifted so keep iterator in place
			i--;
		}
	}
}

int main()
{
	char testStr[13] = "Hello World!";
	printf("String before removal %s", testStr);
	removeCharsFromStr(testStr, 'l', 13);
	printf("String after removal %s", testStr);
    return 0;
}

